app/download_genesis.go

```diff
@@ -0,0 +1,119 @@
+package app
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+)
+
+// genesisBaseURL is the base URL for downloading genesis files. Can be overridden in tests.
+var genesisBaseURL = "https://raw.githubusercontent.com/celestiaorg/networks/master"
+
+// chainIDToSha256 is a map of chainID to the SHA-256 hash of the genesis file for that chain ID.
+// To add a new chain-id, download the genesis file from the networks repo and compute the SHA-256 hash.
+// Add the chain-id and hash to this map.
+var chainIDToSha256 = map[string]string{
+	"celestia":   "9727aac9bbfb021ce7fc695a92f901986421283a891b89e0af97bc9fad187793",
+	"mocha-4":    "0846b99099271b240b638a94e17a6301423b5e4047f6558df543d6e91db7e575",
+	"arabica-10": "fad0a187669f7a2c11bb07f9dc27140d66d2448b7193e186312713856f28e3e1",
+	"arabica-11": "77605cee57ce545b1be22402110d4baacac837bdc7fc3f5c74020abf9a08810f",
```

Comment:
where did you get these hashes btw? 

mocha-4 doesn't seem to be the correct hash: https://github.com/celestiaorg/networks/blob/master/mocha-4/genesis_hash.txt

Reply:
I took them from here:

https://github.com/celestiaorg/celestia-app/blob/e4f57db0a67a326ec9cdced4ccce6a5a595252b3/cmd/celestia-appd/cmd/download_genesis.go#L19-L24